docs/devguides/dependency_mgmt.md

```diff
@@ -0,0 +1,223 @@
+
+   1. Find all `<!-- _RENDERED -->` comments, and for each comment:
+
+   2. Look up the corresponding component class.
+
+   3. Get the component's inlined JS / CSS from `Component.js/css`, and linked JS / CSS from `Component.Media.js/css`.
+
+   4. Generate JS script that loads the JS / CSS dependencies.
+
+   5. Insert the JS scripts either at the end of `<body>`, or in place of `{% component_dependencies %}` / `{% component_js_dependencies %}` tags.
+
+   6. To avoid the [flash of unstyled content](https://en.wikipedia.org/wiki/Flash_of_unstyled_content), we need place the styles into the HTML instead of dynamically loading them from within a JS script. The CSS is placed either at the end of `<head>`, or in place of `{% component_dependencies %}` / `{% component_css_dependencies %}`
+
+   7. We cache the component's inlined JS and CSS, so they can be fetched via an URL, so the inlined JS / CSS an be treated the same way as the JS / CSS dependencies set in `Component.Media.js/css`.
+      - NOTE: While this is currently not entirely necessary, it opens up the doors for allowing plugins to post-process the inlined JS and CSS. Because after it has been post-processed, we need to store it somewhere.
+
+3. Server returns the post-processed HTML.
+
+4. In the browser, the generated JS script from step 2.4 is executed. It goes through all JS and CSS dependencies it was given. If some JS / CSS was already loaded, it is NOT fetched again. Otherwise it generates the corresponding `<script>` or `<link>` HTML tags to load the JS / CSS dependencies.
+
+   In the browser, the "dependency manager JS" may look like this:
+
+   ```js
+   // Load JS or CSS script if not loaded already
+   Components.loadScript("js", '<script src="/abc/xyz/script.js">');
+   Components.loadScript("css", '<link href="/abc/xyz/style.css">');
+
+   // Or mark one as already-loaded, so it is ignored when
+   // we call `loadScript`
+   Components.markScriptLoaded("js", "/abc/def");
+   ```
+
+   Note that `loadScript()` receives a whole `<script>` and `<link>` tags, not just the URL.
+   This is because when Django's `Media` class renders JS and CSS, it formats it as `<script>` and `<link>` tags.
+   And we allow users to modify how the JS and CSS should be rendered into the `<script>` and `<link>` tags.
+   
+   So, if users decided to add an extra attribute to their `<script>` tags, e.g. `<script defer src="http://..."></script>`,
+   then this way we make sure that the `defer` attribute will be present on the `<script>` tag when
+   it is inserted into the DOM at the time of loading the JS script.
+
+5. To be able to fetch component's inlined JS and CSS, django-components adds a URL path under:
+
+   `/components/cache/<str:comp_cls_hash>.<str:script_type>/`
+
+   E.g. `/components/cache/my_table_10bc2c.js/`
+
+   This endpoint takes the component's unique hash, e.g. `my_table_10bc2c`, and looks up the component's inlined JS or CSS.
```

Comment:
The endpoint path looks like a file, but it's not ([see definition](https://github.com/EmilStenstrom/django-components/pull/688/files#diff-371d604ce82d2a01458065f1b137fd981b12af1b0b5c5fb2c1f7f5b45c20003fR720)). We actually take the component's JS / CSS string from the cache.

Reply:
"To be able to fetch component's inlined JS and CSS, django-components adds a URL"... How is the URL connected to INLINE JS/CSS?

Reply:
The endpoint returns the component's inlined JS / CSS.

To understand why:
1. The JS / CSS from `Component.Media.js/css` is formatted by Django as HTML tags. E.g.
    - `<script src="/path/to/my/script.js"></script>`
    - `<link href="/path/to/my/style.css" />`

2. But the inlined JS / CSS is just plain text, e.g.
   ```js
   console.log("hi!")
   ```
   ```css
   .my-table { color: red; }
   ```

3. But we want to allow the client-side dependency manager to 1. Fetch the inlined JS / CSS, and 2. To decide if it needs to be fetched at all.

   To achieve that, the endpoint above returns the INLINED JS / CSS.

4. That way, instead of
   ```js
   console.log("hi!")
   ```
   ```css
   .my-table { color: red; }
   ```

	we can treat the component's inlined JS / CSS also as HTML tags:

    - `<script src="/components/cache/my_table_10bc2c.js"></script>`
    - `<link href="/components/cache/my_table_10bc2c.css" />`